jaeger_client/metrics.py

```diff
@@ -21,69 +21,75 @@
 from __future__ import absolute_import
 
 
-class Metrics(object):
-    """Provides an abstraction of metrics reporting framework."""
+class MetricsFactory(object):
+    """Generates new metrics."""
 
-    # TODO do we want to automatically include host name?
-    prefix = 'jaeger'
+    def __init__(self, count=None, gauge=None, timing=None, tags=None):
+        """
+        :param count: function (key, value) to emit counters
+        :param gauge: function (key, value) to emit gauges
+        :param timing: function (key, value) to emit timings
+        :param tags: {k:v} dictionary
+        """
+        self.count = count
+        self.gauge = gauge
+        self.timing = timing
+        self.tags = tags
 
-    TRACES_STARTED_SAMPLED = '%s.traces-started.sampled' % prefix
-    """Number of traces started by this tracer as sampled"""
+    def counter(self, name, tags=None):
+        return Counter(name, self.merge_tags(tags), self.count)
 
-    TRACES_STARTED_NOT_SAMPLED = '%s.traces-started.not-sampled' % prefix
-    """Number of traces started by this tracer as not sampled"""
+    def timer(self, name, tags=None):
+        return Timer(name, self.merge_tags(tags), self.timing)
 
-    TRACES_JOINED_SAMPLED = '%s.traces-joined.sampled' % prefix
-    """Number of externally started sampled traces this tracer joined"""
+    def gauge(self, name, tags=None):
+        return Gauge(name, self.merge_tags(tags), self.gauge)
 
-    TRACES_JOINED_NOT_SAMPLED = '%s.traces-joined.not-sampled' % prefix
-    """Number of externally started non-sampled traces this tracer joined"""
+    def merge_tags(self, tags=None):
+        if not self.tags:
+            return tags
+        tags_cpy = self.tags.copy()
+        tags_cpy.update(tags)
+        return tags_cpy
 
-    SPANS_SAMPLED = '%s.spans.sampled' % prefix
-    """Number of sampled spans started/finished by this tracer"""
 
-    SPANS_NOT_SAMPLED = '%s.spans.not-sampled' % prefix
-    """Number of not-sampled spans started/finished by this tracer"""
+class Counter(object):
```

Comment:
Have you considered having the factory just return functions, instead of objects? Given the lack of interfaces / compiler checks, functions might be simpler to deal with. It would also be simpler to wrap legacy Metrics as MetricsFactory.